File: ruby/styles/events.rb
module Styles
  class Events < Base
    def leaf?( article)
      false
    end

    def sort( articles)
      no_dates = articles.select {|a| a.date.nil?}
      if no_dates.size > 1
        no_dates.each do |a|
          a.error( "Expected date")
        end
      end

      articles.sort do |a1,a2|
        if a1.date.nil?
          1
        elsif a2.date.nil?
          -1
        else
          (a1.date <=> a2.date)
        end
      end
    end
  end
end

File: java/src/com/alofmethbin/jadawin/styles/Events.java
package com.alofmethbin.jadawin.styles;

import com.alofmethbin.jadawin.Article;
import com.alofmethbin.jadawin.Page;
import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;

public class Events extends BaseStyle {

    @Override
    public boolean isLeaf(Article article) {
        return false;
    }

    @Override
    public void sort( List<Page> children) {
        List<Page> noDates = new ArrayList<>();
        List<Page> dated   = new ArrayList<>();
        
        for (Page child: children) {
            if (child.getDate() == null) {
                noDates.add( child);
            } else {
                dated.add( child);
            }
        }
        
        if (noDates.size() > 1) {
            for (Page child: noDates) {
                child.error( "Expected date");
            }
        }
        
        dated.sort( new Comparator<>() {
            @Override
            public int compare( Page p0, Page p1) {
                return p0.getDate().compareTo( p1.getDate());
            }
        });
        
        children.clear();
        children.addAll( dated);
        children.addAll( noDates);
    }
}
